### PR TITLE
Add pkgconfig file

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/262)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,9 @@
 cmake_minimum_required(VERSION 3.5)
 project(pmemkv)
 
+# set version of pmemkv
+set(VERSION 0.1)
```
Comment (marcinslusarz) [inline CMakeLists.txt]: We need to do something about versioning. It's broken in all of our cmake-based repos.

**Comment by ldorau** [comment]
Rebased on #245

File: CMakeLists.txt [hunk 2]
```diff
@@ -151,21 +130,28 @@ endif()
...
 if(ENGINE_VSMAP OR ENGINE_VCMAP)
 	include(memkind)
+	set(PKG_REQUIRES "${PKG_REQUIRES} memkind")
 endif()
 if(ENGINE_VCMAP OR ENGINE_CMAP)
 	include(tbb)
+	set(PKG_REQUIRES "${PKG_REQUIRES} tbb")
 endif()
 if(ENGINE_CACHING)
 	include(memcached-experimental)
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt]: perhaps add here "XXX" or "TODO" for these packages. When we finally fix or replace them we could make use of pkg conf

**Comment by ldorau** [comment]
@igchor @lukaszstolarczuk @marcinslusarz PR rebased on master, ready for review, please resolve your issues

File: cmake/helpers.cmake [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+function(set_version VERSION)
+	execute_process(COMMAND git describe
```
Comment (marcinslusarz) [inline cmake/helpers.cmake]: Please also handle the VERSION file.

**Comment by ldorau** [comment]
@marcinslusarz please review the last snippet with the `VERSION` file

**Comment by marcinslusarz** [comment]
___
*[cmake/helpers.cmake, line 36 at r9](https://reviewable.io/reviews/pmem/pmemkv/262#-LgRjOAyE49IMFemFabV:-LgRjOAz-YwUPcD3MbPW:b4rvhhm) ([raw file](https://github.com/pmem/pmemkv/blob/bd4097e7afca2306567d5e2b50f4cdb5d78eadbb/cmake/helpers.cmake#L36)):*
> ```CMake
> 	if(EXISTS ${VERSION_FILE})
> 		file(READ ${VERSION_FILE} FILE_VERSION)
> 		# remove all lines but the first one
> ```

VERSION file should have just one line, so I don't see the point of this...

**Pull request merged**